File: api-common/csharp/src/util/ArrayUtil.cs
namespace tr.gov.tubitak.uekae.esya.api.common.util
{
    public static class ArrayUtil
    {
        public static bool Equals(byte[] expected, byte[] actual)
        {
            if (expected == actual)
                return true;

            bool result = ((expected != null) && (actual != null));
            if (result)
            {
                int l = expected.Length;
                result = (l == actual.Length);
                if (result)
                {
                    for (int i = 0; i < l; i++)
                    {
                        if (expected[i] != actual[i])
                            return false;
                    }
                }
            }
            return result;
        }
    }
}

File: api-common/src/tr/gov/tubitak/uekae/esya/api/common/util/ArrayUtil.java
package tr.gov.tubitak.uekae.esya.api.common.util;

import java.util.Arrays;

public class ArrayUtil {

    public static long[] concatArrays(long[] array1, long[] array2) {
        long[] resultArray = Arrays.copyOf(array1, array1.length + array2.length);
        System.arraycopy(array2, 0, resultArray, array1.length, array2.length);
        return resultArray;
    }

    public static <T> T[] concatArrays(T[] array1, T[] array2) {
        T[] result = Arrays.copyOf(array1, array1.length + array2.length);
        System.arraycopy(array2, 0, result, array1.length, array2.length);
        return result;
    }
}
